### PR TITLE
bash: 4.3 -> 4.4

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is WIP. When building, 15 patches are getting fetched and fail - but I generated a patches.nix file which has no patch mentioned... I'm not sure what wents wrong here.
###### Motivation for this change

bash 4.4 has a few new features I'm interested in.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by vcunat** [comment]
@matthiasbeyer: you added a file, but a different one is used (note the version in name).

**Comment by domenkozar** [comment]
I changed the branch to staging, but that breaks things. Can you rebase to staging and change the base?

**Comment by vcunat** [comment]
Staged, after a few more fixes. (Notably: `readline70` was needed for `bashInteractive`.)

**Comment by matthiasbeyer** [comment]
Whops, what happened here? Do I still need to do something or is this subject done here?

**Comment by vcunat** [comment]
It's in the staging branch. GitHub should auto-close the PR when staging gets merged to master.

**Comment by vcunat** [comment]
Hmm, there are nontrivial changes in bash. For now I brought back 4.3 for the non-interactive version and only left 4.4 for `bashInteractive`.

File: pkgs/shells/bash/default.nix [hunk 1]
```diff
@@ -4,11 +4,11 @@ assert interactive -> readline != null;
...
 
 let
-  version = "4.3";
+  version = "4.4";
   realName = "bash-${version}";
-  shortName = "bash43";
+  shortName = "bash44";
   baseConfigureFlags = if interactive then "--with-installed-readline" else "--disable-readline";
-  sha256 = "1m14s1f61mf6bijfibcjm9y6pkyvz6gibyl8p4hxq90fisi8gimg";
+  sha256 = "1jyz6snd63xjn6skk7za6psgidsd53k05cr3lksqybi0q6936syq";
```
Comment (garbas) [inline pkgs/shells/bash/default.nix]: pining @johbo since i know he uses it at rhodecode. do you care to port this patches to 4.3?

Comment (johbo) [inline pkgs/shells/bash/default.nix]: At the moment I think not, since we don't have cygwin active anymore.

**Comment by vcunat** [comment]
I think this PR should be closed, as it seems not to contain any useful code left (but discussion can continue on this thread).

**Comment by Mic92** [comment]
@vcunat do you remember what the non-trivial changes in nix-shell were? I ask because there is a growing mismatch between the bashInteractive (from 2016) and the bash one get, when using `nix-shell` (from 2013). In particular my dotfiles fail.

**Comment by vcunat** [comment]
@Mic92: `nix-shell` changes? I don't understand what you mean (or how it's related to this PR).

**Comment by Mic92** [comment]
@vcunat Sorry. I meant to say: do you remember what the non-trivial changes in bash were, which prevented an upgrade in stdenv?

**Comment by vcunat** [comment]
I don't remember anymore. Maybe it was just to avoid a mass rebuild at that moment.

**Comment by vcunat** [comment]
Hmm, build of `fontconfig-ultimate` got broken by that switch: http://hydra.nixos.org/build/45979225/nixlog/1/raw

**Pull request closed without merging**